src/extern_library_helper.c

```diff
@@ -0,0 +1,39 @@
+#include "extern_library_helper.h"
+#include <dlfcn.h>
+#include <stdlib.h>
+
+LoadedLib load_lib(char* name){
+  // load library
+  void *handle = dlopen(name, RTLD_LAZY);
+  if (!handle) {
+    // error
+  }
+
+  // clear any existing errors
+  dlerror();
+
+  LibraryDefinition (*expose_library) () = dlsym(handle, "expose_library");
+
+  LibraryDefinition lib_def = expose_library();
+
+  ExternFunction* functions = (ExternFunction*) malloc(sizeof(ExternFunction) * lib_def.counter);
+  for (int i = 0; i < lib_def.counter ; i++) {
+    ExternFunction fr = {
+      lib_def.names[i],
+      dlsym(handle, lib_def.names[i])
```

Comment:
This needs more attention: 

What if the user creates the external library in a language that does [name mangling](https://en.wikipedia.org/wiki/Name_mangling), like C++?

The user will not know the symbol name of the functions they want to provide.

Reply:
Now we don't need the function name.